ui/src/plugins/dev.perfetto.AndroidLog/logs_panel.ts

```diff
@@ -109,17 +109,19 @@ export class LogPanel implements m.ClassComponent<LogPanelAttrs> {
 
     this.filterMonitor = new Monitor([() => attrs.filterStore.state]);
 
-    getMachineIds(attrs.trace.engine).then((uMachineIds) => {
-      attrs.cache.uMachineIds = uMachineIds;
-    });
+    if (attrs.cache.uniqueMachineIds === null) {
```

Comment:
If this constructor is called multiple times in quick succession, we could trigger multiple loads.

I realise this isn't the biggest deal as getMachineIds is a very cheap operation seeing as the `cpu` table is small, but I don't think it hurts to be thorough.

If you see https://github.com/google/perfetto/blob/main/ui/src/plugins/dev.perfetto.Ftrace/ftrace_explorer.ts#L113, which does something similar for the ftrace events, we set the cache state to loading when loading to avoid this issue.

TBH, and after thinking a bit about the ftrace one, I'm not quite sure this complexity is entirely worth the effort. It might be better just to load the machine ides in the plugin's onTraceLoad() function and inject them in pre-filled via the attrs.